test/expr/unary/keypath/salvage-with-other-type-errors.swift

```diff
@@ -27,7 +27,7 @@ struct A {
 }
 
 extension A: K {
-    static let j = S(\A.id + "id") // expected-error {{'+' cannot be applied}} expected-note {{}}
+    static let j = S(\A.id + "id") // expected-error {{type of expression is ambiguous without more context}}
```

Comment:
Do you know why did this diagnostic regress so badly? It means that we couldn't diagnose anything at all, which is worrisome to me...

Reply:
I also wonder why does https://github.com/apple/swift/pull/18435/files#diff-5440e49a8b68d1d914483c2c87083645R45 now produce. None of the error messages are checked :(